src/action-status.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const fs = require('fs');
+const crypto = require('crypto');
+const { tracer, eventInterface } = require('epsagon');
+
+let numInvocations = 0;
+const startTime = Date.now();
+const uuid = crypto.randomBytes(16).toString('hex');
+const runningActivations = new Set();
+
+/**
+ * Returns the number of open file handles. currently not needed anymore but can be enabled if
+ * desired.
+ * @param {Logger} log the logger.
+ * @returns {Promise<number>} The number of open file handles.
+ */
+/* istanbul ignore next */
+// eslint-disable-next-line no-unused-vars
+async function getNumOpenFileHandles(log) {
```

Comment:
Why unused ?

Reply:
because it might be expensive to read the directory and we haven't seen a too-many-open-filehandles error for a long time. but I didn't want to discard the code, so we can add it again, if we ever see the error.